src/query/union/simple_union.rs

```diff
@@ -0,0 +1,113 @@
+use crate::docset::{DocSet, TERMINATED};
+use crate::postings::Postings;
+use crate::DocId;
+
+/// A `SimpleUnion` is a `DocSet` that is the union of multiple `DocSet`.
+/// Unlike `BufferedUnion`, it doesn't do any horizon precomputation.
+///
+/// For that reason SimpleUnion is a good choice for queries that skip a lot.
+pub struct SimpleUnion<TDocSet> {
```

Comment:
BTW PISA does that for union and outperforms tantivy and lucene

Reply:
I did some benchmarks some time ago and the intersections + union are faster, but the pure unions are slower with this variant